subprojects/android-test/rx-idler/src/main/kotlin/com/avito/android/test/rxidler/RxIdlingRegistry.kt

```diff
@@ -0,0 +1,110 @@
+package com.avito.android.test.rxidler
+
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.IdlingResource
+import com.squareup.rx3.idler.Rx3Idler
+import io.reactivex.rxjava3.core.Scheduler
+import io.reactivex.rxjava3.functions.Function
+import io.reactivex.rxjava3.functions.Supplier
+import io.reactivex.rxjava3.plugins.RxJavaPlugins
+
+/**
+ * Motivation https://github.com/square/RxIdler/issues/34
+ */
+object RxIdlingRegistry {
+
+    private var initialized = false
+    private var registered = false
+
+    private val registry = mutableListOf<IdlingResource>()
+
+    /**
+     * Must be called one time before Application will be created
+     * read https://github.com/square/RxIdler#usage
+     */
+    fun initialize() {
+        synchronized(RxIdlingRegistry) {
+            require(!initialized) {
+                "Already have been initialized"
+            }
+            initializeInternal()
+            initialized = true
+            registered = true
+        }
+    }
+
+    /**
+     * Enable RX Idling
+     * Should be called only after [disable] have already been called
+     *
+     * [RxIdlingRegistry] is enabled by default after [initialize]
+     */
+    fun enable() {
+        synchronized(RxIdlingRegistry) {
+            require(initialized) {
+                "Must be initialized"
+            }
+            require(!registered) {
+                "Expect registered = false"
+            }
+            IdlingRegistry.getInstance().register(*registry.toTypedArray())
+            registered = true
+        }
+    }
+
+    /**
+     * Disable RX Idling
+     * read [RxIdlingRegistry] motivation
+     *
+     * you should prefer to use [withDisabled] because you could forgot to enable back
+     */
+    fun disable() {
+        synchronized(RxIdlingRegistry) {
+            require(initialized) {
+                "Expect initialized = true"
```

Comment:
```suggestion
                "Must be initialized"
```
To talk in terms of API. It might be easier to understand what to fix.

Reply:
Done